cvmfs/magic_xattr.h

```diff
@@ -0,0 +1,324 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_MAGIC_XATTR_H_
+#define CVMFS_MAGIC_XATTR_H_
+
+#include <map>
+#include <string>
+
+#include "catalog_counters.h"
+#include "directory_entry.h"
+#include "shortstring.h"
+#include "util/exception.h"
+#include "util/string.h"
+
+class MountPoint;
+
+enum MagicXattrFlavor {
```

Comment:
Let's add a comment explaining the relationship of this enum with the MagicXattr class hierarchy